packages/sdk/src/assets/bridge-asset.ts

```diff
@@ -0,0 +1,267 @@
+
+    // now that we have gasLimit and gasPrice, we update the payload to include the proper data needed for estimating fees
+    const bridgeRequestPayload = (
+      await this._createBridgeOutPayload(
+        chainId,
+        holographToChainId,
+        contractAddress,
+        destinationGasLimit,
+        destinationGasPrice,
+        bridgeOutPayload,
+      )
+    ).result
+
+    // we are now ready to get fees for transaction
+    const feeValue = await this._getValue(
+      chainId,
+      holographToChainId,
+      destinationGasLimit,
+      destinationGasPrice,
+      bridgeRequestPayload,
+    )
+
+    const unsignedBridgeOutTx = BridgeAsset.createUnsignedBridgeOutTx(
+      holographToChainId,
+      contractAddress,
+      destinationGasLimit,
+      destinationGasPrice,
+      bridgeOutPayload,
+      //{value: feeValue}, TODO: do we need to pass the value here?
+    )
+
+    // Source transaction parameters
+
+    const sourceGasPrice = this.gasSettings?.sourceGasPrice ?? (await this._providers.byChainId(chainId).getGasPrice())
+
+    const sourceGasLimit = this.gasSettings?.sourceGasLimit ?? BigInt(450000) // NOTE: Only to test user balance
+    /**
+     * TODO: put this in another place, this does not make part of the function scope
+      const userBalance = await this.getSignerBalance()
+      const sourceTxGas = sourceGasPrice.mul(sourceGasLimit)
+      const totalValueNeeded = sourceTxGas.add(feeValue)
+      const error = checkBalanceBeforeTX(userBalance, totalValueNeeded) // TODO: throw error for balance check
+     */
+
+    return {
+      gasSource: {
+        chainId,
+        gasPrice: sourceGasPrice,
+        gasLimit: sourceGasLimit,
+      },
+      gasDestination: {
+        chainId: toChainId,
+        gasPrice: destinationGasPrice,
+        gasLimit: destinationGasLimit,
+      },
+      value: feeValue,
+      unsignedTx: unsignedBridgeOutTx,
+    }
+  }
+
+  protected async _bridgeOut(
+    chainId: number,
+    toChainId: number,
+    wallet: HolographWallet,
+    contractAddress: Address,
+    bridgeOutPayload: Hex,
+    options?: {sourceGasPrice?: bigint; sourceGasLimit?: bigint},
+  ): Promise<TransactionReceipt> {
+    const {unsignedTx, gasSource, value} = await this.estimateBridgeOut(
+      chainId,
+      toChainId,
+      contractAddress,
+      bridgeOutPayload,
+    )
+
+    const walletClient = wallet.onChain(chainId)
+
+    const tx = await walletClient.sendTransaction({
+      chain: walletClient.chain,
+      account: wallet.account,
+      data: unsignedTx,
+      gasPrice: options?.sourceGasPrice ?? gasSource.gasPrice,
+      gas: options?.sourceGasLimit ?? gasSource.gasLimit,
+      value,
+    })
+
+    //@ts-ignore TODO: need to fix the holographWallet _multiChainWalletClient type, since the created WalletClient extend publicActions
+    const receipt = await wallet.waitForTransactionReceipt({hash: tx}) // TODO: should we wait for the transaction to be processed?
```

Comment:
Maybe we should make this an option? Something like `option.shouldWait`? by default its set to false, but if set to true this function will hold.

Reply:
It depends more in what is the goal of the function/tx

On the collection and nft classes (mostly deploying and minting functions), we have to wait because we need to get the tokenId, collection address, and things like that from the receipt. So waiting for the tx is mandatory to decode its receipt

We need to ask ourselves, do we need the tx receipt for the return of `_bridgeOut` function? If not, then we can remove this line and only return the txHash

Reply:
Maybe we don't need an argument for that. It depends more in what is the goal of the function/tx

On the collection and nft classes (mostly deploying and minting functions), we have to wait because we need to get the tokenId, collection address, and things like that from the receipt. So waiting for the tx is mandatory to decode its receipt

We need to ask ourselves, do we need the tx receipt for the return of `_bridgeOut` function? If not, then we can remove this line and only return the txHash

Reply:
@sogoiii  Maybe we don't need an argument for that. It depends more in what is the goal of the function/tx

On the collection and nft classes (mostly deploying and minting functions), we have to wait because we need to get the tokenId, collection address, and things like that from the receipt. So waiting for the tx in those cases is mandatory to decode its receipt

We need to ask ourselves, do we need the tx receipt for the return of `_bridgeOut` function? If not, then we can remove this line and only return the txHash

Reply:
@feliperodriguess @BravoNatalie ok, lets make it wait for the tx to do some of the decoding. We can add options later.